sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -253,9 +253,10 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
    */
   async renewSessionLock(options?: OperationOptions): Promise<Date> {
     this._throwIfReceiverOrConnectionClosed();
-
+    if (!this._messageSession) {
+      throw new Error("Cannot renew the session lock on a non-existing message session.");
```

Comment:
The user has no idea that the message session is non-existing or how it ended up being non existing.
We may need to use the SessionLockLostError instead but only if the session was in use and now it
Check with @ShivangiReja what error type and message they use and use something similar

Also, in Track 1, it was clear when `this._messageSesion` would not be set. But in Track 2, the minute user gets a successful result out of `createSessionReceiver`, this private variable should not be undefined. So, at what point in time would this error be hit?

Reply:
For that same reason, we shouldnt be calling `_createMessageSessionIfDoesntExist` from every operation on the session receiver

Reply:
Checked with Shivangi, they are throwing SessionLockLost for all the session related operations, not just renewSessionLock.
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs#L939

Reply:
Related to #2826

Reply:
Let's update this PR to throw SessionLockLost error when `renewSessionLock` is called and there is no `_messageSession`

Reply:
Updated!

Reply:
I would say the other methods like setState, getState, receiveDeferredMessages, etc should also not create a new message session and throw the same error if there is no `_messageSession`.

Reply:
`_messageSession` had a reason to be undefined before because `createSessionReceiver` was not async and did not actually create the AMQP link until the first call to any of the above methods.

But, now that `createSessionReceiver` is indeed async and does create the AMQP link, I believe a refactor here is in order. How about we update the `SessionReceiverImpl` to take the `messageSession` as input? This way, `messageSession` will not be undefined in any of these methods.

Reply:
I agree to the reasoning, I'll work on the refactoring.